### DISCUSSION TITLE
Debugging Hooks in Payload 3 - Breakpoints Not Working in VSCode - Documentation needed.

### DISCUSSION THREAD

**Discussion by javierlinked** [payloadcms/payload#8281] [Q&A]
Hi everyone,

We're in the process of migrating a large codebase from Payload 2 to Payload 3, which includes custom logic in hooks and endpoints. A key part of our development workflow involves using the debugger, and we need to maintain that functionality.

I've followed [this guide](https://nextjs.org/docs/pages/building-your-application/configuring/debugging) from Next.js and referred to the [Payload 3.0 demo’s launch.json](https://github.com/payloadcms/payload-3.0-demo/blob/main/.vscode/launch.json), but I haven't had much success.

The issue is that VSCode never hits breakpoints in our `hooks` (located in ./src/hooks). Occasionally, it stops at a webpack version of the hook on the first run, but never on the actual source file. Ideally, I'd like to attach to the process/port as we used to with nodemon/ts-node.

Any guidance on how to resolve this would be greatly appreciated!

We are using a blank template with latest beta packages and mongodb adapter.

Thanks.

**Comment by TimBoonstra** [comment]
I'm unable to debug at all under windows :( debugger won't even start, it just throws an error 

Exception has occurred: Error: ENOENT: no such file or directory, stat 'G:\projects\[..project_dir..]\next.COM
  at Object.statSync (node:fs:1668:25)
    at Function.sync (G:\projects\DefiTrader\v2\defi-trader\node_modules\isexe\windows.js:41:23)

**Comment by hugoburguete** [comment]
I am also experiencing this on Payload 3.0. I've tried to add the '--inspect' option, but to no avail. breakpoints never break.

**Comment by immotus** [comment]
The obscure solution is #6882.

The key for me was to adjust the start-up script:

```
"dev": "cross-env NODE_OPTIONS=\"${NODE_OPTIONS} --no-deprecation\" next dev",
```

**Reply by hugoburguete** [reply]
Nice find, it does indeed fix the issue. Thanks.
